pkg/config/store.go

```diff
@@ -78,6 +78,10 @@ type ChangeNotifier interface {
 	// The client should issue ReadChangeLog to see what has changed if the call is available.
 	// else it should re-read the store, perform diff and apply changes.
 	RegisterStoreChangeListener(s StoreListener)
+
+	// IsStoreChangeAvailable returns true only when the backend store is configured
+	// to emit change notifications.
+	IsStoreChangeAvailable() bool
```

Comment:
I think I'm misunderstanding how this is intended to be used.

In my mind:
- `Redis` implements the `ChangeNotifier` interface
- the `manager` registers its own `StoreListener`
- When there's a change to the underlying store, the `Redis` notifies `manager` the registered `StoreListener`
- the `manager`'s `StoreListener` impl calls `Redis`'s `ReadChangeLog` func, retrieving the new configs from the store

is that correct, and if so, how is `IsStoreChangeAvailable` intended to be used?